src/edges_analysis/analysis/beams.py

```diff
@@ -714,6 +717,8 @@ def sky_convolution_generator(
     >>> for i, j, mean_t, conv_t, sky, bm, time, npix in sky_convolution_generator():
     >>>     print(conv_t)
     """
+    if beam_interpolation:
+        beam = beam.at_freq(beam.frequency)
```

Comment:
Hold on.... this shouldn't do anything. The beam is already at `beam.frequency`?

Reply:
@steven-murray added a separate function now to avoid confusion